idaes/core/surrogate/pysmo/sampling.py

```diff
@@ -594,6 +596,13 @@ def __init__(
             self.number_of_samples = number_of_samples
             self.x_data = bounds_array  # Only x data will be present in this case
 
+        if rand_seed is not None:
+            try:
```

Comment:
Running a decimal seed such as `int(15.1)` returns the floored value `15`, while running a non-float seed such as `int("15.1")` raises a `TypeError` - so, neither case will be caught as written here. Should line 602 be `np.random.seed(rand_seed)` and lines 603 nad 604 except and raise a `TypeError`?

Reply:
@bpaul4 One of Andrew's comments (see above) was that the test should pass for floats (1in case someone enters something like 15.0), so it is currently implemented to pass to decimal, non-integer seed values. 

On the error: ``int("15.1")`` raises a ``ValueError`` for me, not a ``TypeError``:
<img width="860" alt="image" src="https://github.com/IDAES/idaes-pse/assets/47539353/83b74efc-b89c-4cf2-a678-c1f9a81c94da">

Reply:
Thanks for clarifying, in that case I think it's OK.

Reply:
Is it raising a TypeError on your end? It would be good to know if it is, before I break the whole of IDAES :)

Reply:
No, you're correct. `np.random.seed("15.1")` throws a `TypeError`.

Reply:
According to the docstring random.seed is a legacy function. Good opportunity to use the best practice interface instead I think.

Reply:
The new best practice interface ``np.random.default_rng()`` requires some code refactoring for it to work in this case. Because the new interface requires the creation of a local random number generator (unlike ``np.random.seed`` which is global), it breaks several of the static methods (e.g.,  in the sampling classes. 

The new method is already used in the case of the ``CustomSampling`` class. Further changes may be more suitable in the future if/when the sampling class is refactored.